iris/iris_test.go

```diff
@@ -0,0 +1,71 @@
+package iris
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/segmentio/encoding/json"
+	"syreclabs.com/go/faker"
+)
+
+type jSMock struct {
+	Name    string   `json:"name"`
+	Company string   `json:"company"`
+	Emails  []string `json:"emails"`
+}
+
+func TestGetResponse(t *testing.T) {
+	name := faker.Name().FirstName()
+	emails := []string{faker.Internet().Email(), faker.Internet().Email()}
+
+	t.Run("It should decode response", func(t *testing.T) {
+		b, err := json.Marshal(map[string]interface{}{
+			"data": map[string]interface{}{"name": name},
+		})
+		if err != nil {
+			panic(err)
+		}
+
+		res := http.Response{
+			Body: ioutil.NopCloser(bytes.NewBuffer(b)),
+		}
+
+		var data jSMock
+		err = GetResponse(&res, &data)
+		if err != nil {
+			t.Fatal(errors.Wrap(err, "error getting response"))
+		}
+
+		if data.Name != name {
+			t.Fatalf("expected %s got %s", name, data.Name)
+		}
+	})
+
+	t.Run("It should decode array response", func(t *testing.T) {
+		b, err := json.Marshal(map[string]interface{}{
+			"data": map[string]interface{}{
+				"emails": emails,
```

Comment:
once again you're not really testing that it can parse an array. As `data` is mapped to another map that wraps the array

Reply:
perhaps I'm getting the wrong idea of what the test is aiming for. If it's to test if it can ready array properties directly from `data` it doesn't test that. If it's otherwise to test whether `mapstructure` can read array properties from a `data` map then okay. Although the second test is redundant as I'm sure `mapstructure` has such test.